src/values/ObjectValue.js

```diff
@@ -230,6 +236,9 @@ export default class ObjectValue extends ConcreteValue {
   // partial objects
   _isPartial: BooleanValue;
 
+  // tainted objects
+  _isTainted: BooleanValue;
```

Comment:
I think there's a conceptual issue in Prepack around these tracked internal properties (https://github.com/facebook/prepack/issues/1124): The formal type should really be `BooleanValue | AbstractValue`, as merging diverged control-flow paths where a value became tainted on one side, but not on the other, would result in a conditionally tainted value. Same issue applies to `_isPartial` if applied later after construction. So, while not a new issue, we shouldn't compound the problem going forward. I think this may add a big new layer of complexity to the implementation and reasoning...

Reply:
Yea, I'm not really sure about what to do about this one. It is nice to avoid having to taint other branches when only one is unknown. However, if we keep this diverged path, we need to deal with it in the Binding case too.